pkg/cmd/render/render.go

```diff
@@ -0,0 +1,162 @@
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+type renderOpts struct {
+	ppInputFiles ppFiles
+	assetsInDir  string
+	assetsOutDir string
+}
+
+type ppFiles []string
+
+func (ppf *ppFiles) String() string {
+	return fmt.Sprint(*ppf)
+}
+
+func (ppf *ppFiles) Type() string {
+	return "ppFiles"
+}
+
+// Set parse performance-profile-input-files flag and store it in ppf
+func (ppf *ppFiles) Set(value string) error {
+	if len(*ppf) > 0 {
+		return errors.New("performance-profile-input-files flag already set")
+	}
+
+	for _, s := range strings.Split(value, ",") {
+		*ppf = append(*ppf, s)
+	}
+	return nil
+}
+
+//NewRenderCommand creates a render command.
+func NewRenderCommand() *cobra.Command {
+	renderOpts := renderOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "render",
+		Short: "Render performance-addon-operator manifests",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := renderOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Var(&r.ppInputFiles, "performance-profile-input-files", "A comma-separated list of performance-profile manifests.")
+	fs.StringVar(&r.assetsInDir, "asset-input-dir", components.AssetsDir, "Input path for the assets directory.")
+	fs.StringVar(&r.assetsOutDir, "asset-output-dir", r.assetsOutDir, "Output path for the rendered manifests.")
+	// environment variables has precedence over standard input
+	r.readFlagsFromEnv()
+}
+
+func (r *renderOpts) readFlagsFromEnv() {
+	if ppInFiles := os.Getenv("PERFORMANCE_PROFILE_INPUT_FILES"); len(ppInFiles) > 0 {
+		r.ppInputFiles.Set(ppInFiles)
+	}
+
+	if assetInDir := os.Getenv("ASSET_INPUT_DIR"); len(assetInDir) > 0 {
+		r.assetsInDir = assetInDir
+	}
+
+	if assetsOutDir := os.Getenv("ASSET_OUTPUT_DIR"); len(assetsOutDir) > 0 {
+		r.assetsOutDir = assetsOutDir
+	}
+}
+
+func (r *renderOpts) Validate() error {
+	if len(r.ppInputFiles) == 0 {
+		return fmt.Errorf("performance-profile-input-files must be specified")
+	}
+
+	if len(r.assetsOutDir) == 0 {
+		return fmt.Errorf("asset-output-dir must be specified")
+	}
+
+	return nil
+}
+
+func (r *renderOpts) Run() error {
+	for idx, pp := range r.ppInputFiles {
+		b, err := ioutil.ReadFile(pp)
+		if err != nil {
+			return err
+		}
+
+		profile := &performancev2.PerformanceProfile{}
+		err = yaml.Unmarshal(b, profile)
+		if err != nil {
+			return err
+		}
+
+		components, err := controllers.GetNewComponents(profile, &r.assetsInDir)
+		if err != nil {
+			return err
+		}
+		or := []v1.OwnerReference{
+			{
+				Kind: profile.Kind,
+				Name: profile.Name,
+			},
+		}
+
+		for name, component := range components {
+			component.(v1.Object).SetOwnerReferences(or)
+			b, err := yaml.Marshal(component)
```

Comment:
@fromanirh  Any idea how to marshal here in a loop? converting to object isn't enough

Reply:
I will have a closer look later